### ISSUE TITLE
Data Caching approach in SQL Lite

### ISSUE THREAD

**Issue by prasanna-jm** [RevealBi/Reveal.Sdk#254]
1) We need solution to protect our sensitive data stored in SQL Lite.

2) During our development, we discovered that even when selecting the option to always refresh data, Reveal BI still stores the data in cache, expected no caching when we select always refresh the data.

**Comment by brianlagunas** [comment]
What are you asking for exactly? What do you mean by "a solution to protect sensitive data in SQLite"? Protect how? Are you also asking for a way to turn off caching? These two items seem to contradict each other. Could you please clarify your request with as much detail as possible.

**Comment by prasanna-jm** [comment]
1)	We require a solution to ensure the protection of our sensitive data stored in SQL Lite.
**Detailed Explanation:** Currently, Reveal BI caches data in sql lite files as plain text which can be read by person having access sql lite db path, which poses a significant concern for sensitive information. While this approach may be suitable for some implementations, it becomes problematic when dealing with Personally Identifiable Information (PII) or other highly sensitive data. In some cases, even administrators should not have access to this type of data. To address this issue, we need data protection, such as implementing encryption and decryption measures.

2)	We have identified an issue during our development process regarding data caching in Reveal BI, even when the “always refresh data” option is selected.
**Detailed Explanation:** When creating a dashboard, we have the option to choose “always refresh data,” implying that the data should not be stored in the cache. However, we have observed that Reveal BI still caches the data, which is not the expected behavior.
To address this concern, we are seeking a way to disable or turn off caching when the “always refresh data” option is selected. This functionality would be particularly valuable when processing PII data, as it would help us ensure that sensitive information is not persistently stored in the cache.

 
**In summary, we are seeking solutions for both encrypting and decrypting data during the caching process, as well as the ability to disable caching when the “always refresh data” option is selected. These measures will enable us to enhance data security and privacy within the Reveal BI platform.**

**Comment by brianlagunas** [comment]
This has been implemented. To enable encryption use the following code
```
builder.Services.AddControllers().AddReveal(builder=>
{
    builder.AddSettings(settings => 
    {
        settings.EnableEncryption("password");
    });
});
```
